model/aro_hcp/v1alpha1/managed_identities_requirements_type.model

```diff
@@ -0,0 +1,116 @@
+  ControlPlaneOperators []ControlPlaneOperatorIdentity
+  // The data plane operators managed identities requires
+  DataPlaneOperators []DataPlaneOperatorIdentity
+}
+
+struct ControlPlaneOperatorIdentity {
+  // indicates the whether the identity is always required or not.
+  // "always" means that the identity is always required
+  // "on_enablement" means that the identity is only required when a functionality 
+  // that leverages the operator is enabled
+  Required OperatorIdentityRequirement
+
+   // the name of the control plane operator that needs the identity
+   Name String
+
+   // The field is a string and it is of format X.Y.
+   // Not specifying it indicates support for this operator in all Openshift versions,
+   // starting from min_openshift_version if min_openshift_version is defined. 
+   @json(name="max_openshift_version")
+   MaxOpenShiftVersion String
+
+   // The field is a string and it is of format X.Y.
+   // Not specifying it indicates support for this operator in all Openshift versions,
+   // or up to max_openshift_version, if defined. 
+   @json(name="min_openshift_version")
+   MinOpenShiftVersion String   
+
+   // A list of roles that are required by the operator 
+   RoleDefinitions []RoleDefinition
+}
+
+struct DataPlaneOperatorIdentity {
+   // indicates the whether the identity is always required or not
+   // "always" means that the identity is always required
+   // "on_enablement" means that the identity is only required when a functionality 
+   // that leverages the operator is enabled
+   Required OperatorIdentityRequirement
+
+   // the name of the data plane operator that needs the identity
+   Name String
+
+   // The field is a string and it is of format X.Y.
+   // Not specifying it indicates support for this operator in all Openshift versions,
+   // starting from min_openshift_version if min_openshift_version is defined. 
+   @json(name="max_openshift_version")
+   MaxOpenShiftVersion String
+
+   // The field is a string and it is of format X.Y.
+   // Not specifying it indicates support for this operator in all Openshift versions,
+   // or up to max_openshift_version, if defined. 
+   @json(name="min_openshift_version")
+   MinOpenShiftVersion String
+
+   // It is a list of K8s ServiceAccounts leveraged by the operator.
+   // There must be at least a single service account specified.
+   // This information is needed to federate a managed identity to a k8s subject.
+   // There should be no duplicated "name:namespace" entries within this field. 
+   ServiceAccounts []IdentityServiceAccount
+
+   // A list of roles that are required by the operator 
+   RoleDefinitions []RoleDefinition
+}
+
+struct IdentityServiceAccount {
+  // The name of the service account to be leveraged by the operator
+  Name String
+  // The namespace of the service account to be leverage by the operator
+  Namespace String
+}
+
+enum OperatorIdentityRequirement {
```

Comment:
According to the guidelines, this shouldn't be an enum. Rather a string that we will validate at app level

Reply:
yes

Reply:
Changed it to a string with the possible values documented